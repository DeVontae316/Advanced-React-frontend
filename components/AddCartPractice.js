import React from 'react';
import PropTypes from 'prop-types';
import gql from 'graphql-tag';
import {Mutation} from 'react-apollo';
import Error from './ErrorMessage';
import {CURRENT_USER_QUERY} from './User';

const ADD_TO_CART =  gql`
mutation ADD_TO_CART($id:ID!){
    addToCart(id:$id){
        id 
        quantity
       
        
    }
}
`


const AddCartPractice = ({item})=>{
    return(
        <Mutation 
       
       refetchQueries={[{query:CURRENT_USER_QUERY}]}
        mutation={ADD_TO_CART} 
        variables={{id:item}}>
            {(addToCart,{loading,error})=>{
            
            if(error) return <Error error={error} />
             return <button disabled={loading} onClick={addToCart}>
                 Add To Cart Practice
             </button>

             }}
        </Mutation>
       
    )
}


export default AddCartPractice;
export{ADD_TO_CART}

