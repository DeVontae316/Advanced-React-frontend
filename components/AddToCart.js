import React,{Component} from 'react';
import {Mutation} from 'react-apollo';
import gql from 'graphql-tag';

const ADD_TO_CART = gql`
 mutation ADD_TO_CART($id:ID!){
     addToCart(id:$id){
         id
         quantity 
     }
 }
`

class AddToCart extends Component{
    render(){
        const{id} = this.props;
        return (<Mutation  mutation={ADD_TO_CART}variables={{id:id}}> 
              {(addToCart,{loading,error})=>(
                  <button onClick={addToCart}>Add Item To Cart</button>
              )}
            </Mutation>
        )
        
    }
}


export default AddToCart;
export{ADD_TO_CART};