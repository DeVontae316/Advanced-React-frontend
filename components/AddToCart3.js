import React from 'react';''
import{Mutation} from 'react-apollo';
import gql from 'graphql-tag';
import PropTypes from 'prop-types';
import {CURRENT_USER_QUERY} from './User';

const ADD_ITEM_MUTATION = gql`
 mutation ADD_ITEM_MUTATION($id:ID!){
     addToCart(id:$id){
         id 
         quantity 
         user{
             id 
             email 
             name
         }
     }
 }
`;

const AddToCart3 = (props)=>{
    return(
       <Mutation refetchQueries={[{query:CURRENT_USER_QUERY}]} variables={{id:props.id}}mutation={ADD_ITEM_MUTATION}>
           {(addToCart,{loading,error})=>(
               <button onClick={addToCart}>ADD TO CART</button>
           )}
       </Mutation> 
        
    )
}


export  default AddToCart3;