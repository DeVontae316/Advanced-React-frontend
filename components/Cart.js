 import React from 'react';
 import CartStyles  from './styles/CartStyles';
 import Supreme  from './styles/Supreme';
 import CloseButton  from './styles/CloseButton';
 import SickButton from './styles/SickButton';
 import User from './User';
 import {Query,Mutation} from 'react-apollo';
 import gql from 'graphql-tag';
 import CartItem from './CartItem';
 import calcTotalPrice from '../lib/calcTotalPrice';
 import formatMoney from '../lib/formatMoney';
 import TakeMyMoney from './TakeMyMoney';

 const LOCAL_STATE_QUERY = gql`
  query{
      cartOpen @client
  }
 `
 const LOCAL_STATE_MUTATION = gql`
  mutation{
      toggleCart @client
  }
 `

const Cart = ()=>(
    /*<div>
        <p>Test</p>
    </div>*/
   <User>
      {({data:{me}})=>{
          if(!me) return null;
          
          
         return <Mutation mutation={LOCAL_STATE_MUTATION}>
         {(toggleCart,{loading,error})=>(
            <Query query={LOCAL_STATE_QUERY}>
        
            {({data,error,loading})=> console.log(me) || (
                <CartStyles open ={data.cartOpen}>
                <header>
                    <CloseButton onClick={toggleCart} title="close">&times;</CloseButton>
                    <Supreme>Name</Supreme>
                  <p>You have {me.cart.length} item in your cart</p>
                    
                </header>
                 <ul>
                 {me.cart.map((cartItem,index,array)=>
                 <CartItem key={cartItem.id}cartItem={cartItem} />)}
                
          
                  </ul>
                <footer><p>Total:{formatMoney(me.cart.reduce((reducer,cartItem)=>{
                        
                        return reducer  + cartItem.quantity * cartItem.item.price;
                },0))}</p>
                 <TakeMyMoney>
                   {me.cart.length ? <SickButton>Checkout</SickButton> : <h3>Please buy something:)</h3>}
                 </TakeMyMoney>
                 
                </footer>
        
            </CartStyles>
            )} 
         </Query>
         )}
     </Mutation>
      }}
   </User>
    
       
    )

    


export default Cart;
export {LOCAL_STATE_QUERY,LOCAL_STATE_MUTATION};

/*
Skills:
Find total items in the User component me.cart array

Code a coditional that 
will change the word item to items if the length 
from me.cart is > 1

Know how to use the array method reduce to add a total value
*/