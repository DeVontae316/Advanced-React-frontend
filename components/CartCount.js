import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import{TransitionGroup, CSSTransition} from 'react-transition-group';

const Dot = styled.div`
background:${props => props.theme.red};
border-radius:50%;
padding:0.5rem;
line-height:2rem;
min-width:3rem;
margin-left:1rem;
font-weight:100;
font-feature-setting:'tnum';
font-variant-numeric:tabular-numeric;
.count{
    color:white;
    text-align:center;
}
`
const AnimationStyles = styled.span`
 position:relative;
 .count{
     display:block;
     position:relative;
     transition:all 0.2s;
     backface-visibility:hidden;

 }
 .count-enter{
     transform: scale(1) rotateX(0.4turn);
 }
 .count-enter-active{
    transform: rotateX(0);
}
.count-exit{
 top:10;
 positon:relative;
 transform:rotateX(0);
}
.count-exit-active{
    transform: scale(1) rotateX(0.4turn);
}
`
const CartCount = ({count})=>{
    return(
    <AnimationStyles>
        <TransitionGroup>
            <CSSTransition 
            unmountOnExit
            className="count" 
            classNames="count" 
            key={count}
            timeout={{enter:200,exit:200}}
            >
            <Dot>
                <div className="count">
                    {count}
                </div>
            </Dot>
            </CSSTransition>
        </TransitionGroup>
    </AnimationStyles>

    
    )
}



export default CartCount;