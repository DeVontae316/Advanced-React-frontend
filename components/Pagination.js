import React from 'react';
import PaginationStyles from './styles/PaginationStyles';
import gql from 'graphql-tag';
import {Query} from 'react-apollo';
import {perPage} from '../config';
import Head from 'next/head';
import Link from 'next/link';
import styled from 'styled-components';



const Center = styled.div`
 text-align:center;
 margin-top:8px;
 a:hover{
   cursor:alias;
 }
`

const PAGINATION_QUERY = gql`
query PAGINATION_QUERY{
  itemsConnection{
    aggregate{
      count
    }
  }
}
`;

const Pagination = props =>{
  return(
    <Query query={PAGINATION_QUERY}>
     {({data,loading,error})=>{
       const totalItems = data.itemsConnection.aggregate.count;
       const page = props.page;
       const pages = Math.ceil(totalItems / perPage);
       if(loading)<p>...loading</p>
       if(error)<p>{error.message}</p>
       return<PaginationStyles>

         <Link href={{pathname:"/items",query:{page:page - 1}}}>
          <a aria-disabled={page <= 1}>Prev</a>
         </Link>

          <p>Page {props.page} out of {pages}</p>

         <Link href={{pathname:"/items",query:{page:page + 1}}}>
           <a aria-disabled={page >= pages}>Next</a>
         </Link>

          <p>{totalItems} items total</p>
       </PaginationStyles>
     }}
    </Query>
  )
}

export{PAGINATION_QUERY};
export default Pagination;
