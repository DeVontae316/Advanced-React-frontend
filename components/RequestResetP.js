import React,{Component} from 'react';
import Form from './styles/Form';
import gql from 'graphql-tag';
import {Mutation} from 'react-apollo';
import ErrorMessage from './ErrorMessage';
//1.Create a form that will recieve an email to recieve an resetToken
/*2.Create an graphql mutation via apollo that will
recieve the users email*/
const REQUEST_TOKEN_MUTATION = gql`
 mutation REQUEST_TOKEN_MUTATION($email:String!){
   requestToken(email:$email){
     message
   }
 }
`
class RequestReset extends Component {
  state = {
    email:" ",
  }

  onChange = (e)=>{
    const{name,type,value} = e.target;

    this.setState({
      [name]:value
    });
  }
  render(){
    return(
      <Mutation mutation={REQUEST_TOKEN_MUTATION} variables={this.state}>
       {(requestToken,{loading,error,called})=>{
         if(loading)<p>loading...</p>
         return<Form method="post"
         data-test="form"
         onSubmit={async (e)=>{
           
           e.preventDefault();
           await requestToken();
           this.setState({
             email: " ",
           });
         }}>
          CHANGE PASSWORD
          <ErrorMessage error={error} />
          {!error && !loading && called && <p>Check email for reset link</p>}
          <fieldset>

           <span>EMAIL</span>
           <label htmlFor="email">
             <input type="text"
             onChange={this.onChange}
             placeholder="Email"
             required value = {this.state.email}
             id="email"
             name="email"/>
           </label>
           <button type="onSubmit">Change Password</button>
          </fieldset>
         </Form>
       }}
      </Mutation>

    )
  }
}
export{REQUEST_TOKEN_MUTATION };
export default RequestReset;
