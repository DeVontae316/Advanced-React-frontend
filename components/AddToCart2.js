import React from 'react';
import gql from 'graphql-tag';
import{Mutation} from 'react-apollo';
import PropTypes from 'prop-types';
import AddToCart from './AddToCart';
import {CURRENT_USER_QUERY} from './User';

const ADD_TO_CART = gql`
 mutation ADD_TO_CART($id:ID!){
     addToCart(id:$id){
         id 
         quantity
         user{
             id 
             name
         }
     }
 }
`

const AddToCart2 = (props)=>{
    return(
        <Mutation refetchQueries={[{query:CURRENT_USER_QUERY}]} mutation={ADD_TO_CART} variables={{id:props.id}}>
           {(addToCart,{loading,error})=>(
               <button disabled={loading} onClick={addToCart}>ADD{loading && 'ing'} ITEM</button>
           )}
        </Mutation>
    )
}


export default AddToCart2;