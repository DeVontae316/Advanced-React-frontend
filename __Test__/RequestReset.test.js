import Pagination,{PAGINATION_QUERY} from "../components/Pagination";
import {mount} from 'enzyme';
import wait from 'waait';
import {MockedProvider} from 'react-apollo/test-utils';
import {fakeItem, fakeUser} from '../lib/testUtils';
import toJSON from 'enzyme-to-json';
import{REQUEST_TOKEN_MUTATION } from '../components/RequestResetP';
import RequestResetP from '../components/RequestResetP';
//Stop @ 5:15
const mocks = [{
    request:{
        query:REQUEST_TOKEN_MUTATION,
        variables:{
            email:"Test2@gmail.com"
        },
        result:{
            data:{
                requestToken:{
                    message:"Success",
                    __typename:"Message"
                }
            }
        }
    }
}]

describe("<Request/>",()=>{
    fit("Renders and matches snap shot",async()=>{
        const wrapper = mount(
            <MockedProvider >
             <RequestResetP />
            </MockedProvider>
        )
        const form = wrapper.find("form[data-test='form']");
       
        expect(toJSON(form)).toMatchSnapshot();
    });
   
})