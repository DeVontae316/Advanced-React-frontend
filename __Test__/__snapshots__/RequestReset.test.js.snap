// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Request/> Renders and matches snap shot 1`] = `
<form
  className="Form-sc-1xszr8q-0 krRxky"
  data-test="form"
  method="post"
  onSubmit={[Function]}
>
  CHANGE PASSWORD
  <DisplayError
    error={Object {}}
  />
  <fieldset>
    <span>
      EMAIL
    </span>
    <label
      htmlFor="email"
    >
      <input
        id="email"
        name="email"
        onChange={[Function]}
        placeholder="Email"
        required={true}
        type="text"
        value=" "
      />
    </label>
    <button
      type="onSubmit"
    >
      Change Password
    </button>
  </fieldset>
</form>
`;
